src/main/java/org/sagebionetworks/bridge/spring/filters/MetricsFilter.java

```diff
@@ -61,10 +75,21 @@ public void doFilter(ServletRequest req, ServletResponse res, FilterChain chain)
 
         metrics.setQueryParams(paramsMap);
 
+        // Used for record metrics from the UserSession.
+        UserSession session = null;
+        // Log session info when a session is present
         try {
             chain.doFilter(req, res);
             metrics.setStatus(response.getStatus());
+            session = (UserSession) request.getAttribute("CreatedUserSession");
+        } catch (ConsentRequiredException e) {
+            session = e.getUserSession();
+            throw e;
```

Comment:
There's no need to catch the ConsentRequiredException here. We've already set the attribute at the Controllers level.

Reply:
Or the inverse, do it here but don't do it multiple places in the controllers.